text_extensions_for_pandas/array/arrow_conversion.py

```diff
@@ -0,0 +1,327 @@
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+#
+# arrow_conversion.py
+#
+# Part of text_extensions_for_pandas
+#
+# Provide Arrow compatible classes for serializing to pyarrow.
+#
+
+import numpy as np
+import pyarrow as pa
+
+from text_extensions_for_pandas.array import CharSpanArray, TokenSpanArray
+
+
+class ArrowCharSpanType(pa.PyExtensionType):
+    """
+    PyArrow extension type definition for conversions to/from CharSpan columns
+    """
+
+    TARGET_TEXT_KEY = b"target_text"  # metadata key/value gets serialized to bytes
+    BEGINS_NAME = "char_begins"
+    ENDS_NAME = "char_ends"
+
+    def __init__(self, index_dtype, target_text):
+        """
+        Create an instance of a CharSpan data type with given index type and
+        target text that will be stored in Field metadata.
+
+        :param index_dtype:
+        :param target_text:
+        """
+        assert pa.types.is_integer(index_dtype)
+
+        # Store target text as field metadata
+        metadata = {self.TARGET_TEXT_KEY: target_text}
+
+        fields = [
+            pa.field(self.BEGINS_NAME, index_dtype, metadata=metadata),
+            pa.field(self.ENDS_NAME, index_dtype)
+        ]
+
+        pa.PyExtensionType.__init__(self, pa.struct(fields))
+
+    def __reduce__(self):
+        index_dtype = self.storage_type[self.BEGINS_NAME].type
+        metadata = self.storage_type[self.BEGINS_NAME].metadata
+        return ArrowCharSpanType, (index_dtype, metadata)
+
+
+class ArrowTokenSpanType(pa.PyExtensionType):
+    """
+    PyArrow extension type definition for conversions to/from TokenSpan columns
+    """
+
+    TARGET_TEXT_KEY = ArrowCharSpanType.TARGET_TEXT_KEY
+    BEGINS_NAME = "token_begins"
+    ENDS_NAME = "token_ends"
+
+    def __init__(self, index_dtype, target_text):
+        """
+        Create an instance of a TokenSpan data type with given index type and
+        target text that will be stored in Field metadata.
+
+        :param index_dtype:
+        :param target_text:
+        """
+        assert pa.types.is_integer(index_dtype)
+
+        # Store target text as field metadata
+        metadata = {self.TARGET_TEXT_KEY: target_text}
+
+        fields = [
+            pa.field(self.BEGINS_NAME, index_dtype),
+            pa.field(self.ENDS_NAME, index_dtype),
+            pa.field(ArrowCharSpanType.BEGINS_NAME, index_dtype, metadata=metadata),
+            pa.field(ArrowCharSpanType.ENDS_NAME, index_dtype)
+        ]
+
+        pa.PyExtensionType.__init__(self, pa.struct(fields))
+
+    def __reduce__(self):
+        index_dtype = self.storage_type[self.BEGINS_NAME].type
+        metadata = self.storage_type[ArrowCharSpanType.BEGINS_NAME].metadata
+        return ArrowTokenSpanType, (index_dtype, metadata)
+
+
+def char_span_to_arrow(char_span):
```

Comment:
Missing type hints here. Should be `def char_span_to_arrow(char_span) -> pyarrow.ExtensionArray:`